src/conversions/chrono.rs

```diff
@@ -0,0 +1,926 @@
+        let date = PyDate::new(py, self.year(), month, day).expect("Failed to construct date");
+        date.into()
+    }
+}
+
+impl IntoPy<PyObject> for NaiveDate {
+    fn into_py(self, py: Python<'_>) -> PyObject {
+        ToPyObject::to_object(&self, py)
+    }
+}
+
+impl FromPyObject<'_> for NaiveDate {
+    fn extract(ob: &PyAny) -> PyResult<NaiveDate> {
+        let date = <PyDate as PyTryFrom>::try_from(ob)?;
+        Ok(NaiveDate::from_ymd(
+            date.get_year(),
+            date.get_month() as u32,
+            date.get_day() as u32,
+        ))
+    }
+}
+
+impl ToPyObject for NaiveTime {
+    fn to_object(&self, py: Python<'_>) -> PyObject {
+        let h = self.hour() as u8;
+        let m = self.minute() as u8;
+        let s = self.second() as u8;
+        let ns = self.nanosecond();
+        let (ms, fold) = match ns.checked_sub(1_000_000_000) {
+            Some(ns) => (ns / 1000, true),
+            None => (ns / 1000, false),
+        };
+        let time =
+            PyTime::new_with_fold(py, h, m, s, ms, None, fold).expect("Failed to construct time");
+        time.into()
+    }
+}
+
+impl IntoPy<PyObject> for NaiveTime {
+    fn into_py(self, py: Python<'_>) -> PyObject {
+        ToPyObject::to_object(&self, py)
+    }
+}
+
+impl FromPyObject<'_> for NaiveTime {
+    fn extract(ob: &PyAny) -> PyResult<NaiveTime> {
+        let time = <PyTime as PyTryFrom>::try_from(ob)?;
+        let ms = time.get_fold() as u32 * 1_000_000 + time.get_microsecond();
+        let h = time.get_hour() as u32;
+        let m = time.get_minute() as u32;
+        let s = time.get_second() as u32;
+        Ok(NaiveTime::from_hms_micro(h, m, s, ms))
+    }
+}
+
+impl ToPyObject for NaiveDateTime {
+    fn to_object(&self, py: Python<'_>) -> PyObject {
+        let date = self.date();
+        let time = self.time();
+        let yy = date.year();
+        let mm = date.month() as u8;
+        let dd = date.day() as u8;
+        let h = time.hour() as u8;
+        let m = time.minute() as u8;
+        let s = time.second() as u8;
+        let ns = time.nanosecond();
+        let (ms, fold) = match ns.checked_sub(1_000_000_000) {
+            Some(ns) => (ns / 1000, true),
+            None => (ns / 1000, false),
+        };
+        let datetime = PyDateTime::new_with_fold(py, yy, mm, dd, h, m, s, ms, None, fold)
+            .expect("Failed to construct datetime");
+        datetime.into()
+    }
+}
+
+impl IntoPy<PyObject> for NaiveDateTime {
+    fn into_py(self, py: Python<'_>) -> PyObject {
+        ToPyObject::to_object(&self, py)
+    }
+}
+
+impl FromPyObject<'_> for NaiveDateTime {
+    fn extract(ob: &PyAny) -> PyResult<NaiveDateTime> {
+        let dt = <PyDateTime as PyTryFrom>::try_from(ob)?;
+        let h = dt.get_hour().into();
+        let m = dt.get_minute().into();
+        let s = dt.get_second().into();
+        let ms = dt.get_microsecond();
+        let dt = NaiveDateTime::new(
```

Comment:
Right, given we are enforcing a 1:1 translation, returning a hard error here rather than removing tzinfo (or doing a conversion assuming local timezone) seems the reasonable thing to do :+1:

Reply:
I'll add a test too

Reply:
I think it's the right approach, it means that if a user really wants to convert a timezone aware datetime to a naive one, he has to remove the tzinfo himself in the python side.

Reply:
I also added a test